File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/models/TestResult.java
package com.timsmithsoftware.integration_tests.models;

public class TestResult {
	public TestResult(boolean result) {
		this.result = result;
	}

	public boolean result;

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }

        if (obj.getClass() != this.getClass()) {
            return false;
        }
        
        final TestResult other = (TestResult) obj;
        return other.result == this.result;
    }
}

File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/models/TestResult.kt
package com.timsmithsoftware.integration_tests.models

class TestResult (var resultNotes: String = ""){
    companion object {
        private lateinit var failureReason: String
        fun FALSE(reason: String): TestResult {
            this.failureReason = reason
            return TestResult(resultNotes = reason)
        }

        val TRUE: TestResult = TestResult()
    }

}